python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py

```diff
@@ -109,28 +95,25 @@ async def complete_async(
             )
 
             completions = list()
-            if self._task == "text-generation" or self._task == "text2text-generation":
+            if self.task == "text-generation" or self.task == "text2text-generation":
                 for response in results:
                     completions.append(response["generated_text"])
                 if len(completions) == 1:
                     return completions[0]
-                else:
-                    return completions
+                return completions
 
-            elif self._task == "summarization":
+            if self.task == "summarization":
                 for response in results:
                     completions.append(response["summary_text"])
                 if len(completions) == 1:
                     return completions[0]
-                else:
-                    return completions
+                return completions
 
-            else:
-                raise AIException(
-                    AIException.ErrorCodes.InvalidConfiguration,
-                    "Unsupported hugging face pipeline task: only \
-                        text-generation, text2text-generation, and summarization are supported.",
-                )
+            raise AIException(
```

Comment:
nit: wouldn't it be better to do this check before we try to do any work? If the task is not defined properly, why spend resources on anything? Perhaps we don't need to fix now, but we could in a next iteration.